### PR TITLE
Possible improvements

### PR DESCRIPTION
**Pull request by makcandrov**
Here are some possible improvements that I came across while writing the solutions

### CODE REVIEW

File: src/Hack.sol [hunk 1]
```diff
@@ -39,21 +39,20 @@ contract Store {
 
 /* Exercise 2 */
 
-/// @dev You can buy some objects.
-///      Further purchases are discounted.
-///      You need to pay basePrice / (1 + objectBought), where objectBought is the number of objects you previously bought.
+/// @dev Enables users to buy objects at discounted prices.
 contract DiscountedBuy {
-    uint256 public basePrice = 1 ether;
+    uint256 public constant basePrice = 1 ether;
```
Comment (MerlinEgalite) [inline src/Hack.sol:44]:
```suggestion
    uint256 public constant BASE_PRICE = 1 ether;
```

File: src/Hack.sol [hunk 2]
```diff
@@ -116,19 +115,19 @@ contract Vault {
 
 /// @dev Contract for locking and unlocking funds using a commitment and password.
 contract Locker {
-    bytes32 commitment;
+    bytes32 _commitment;
```
Comment (MerlinEgalite) [inline src/Hack.sol:118]:
```suggestion
    bytes32 internal _commitment;
```

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]
Nice!

File: src/Hack.sol [hunk 3]
```diff
@@ -457,8 +453,6 @@ contract GuessTheAverage {
     mapping(address => Player) public players; // Maps an address to its respective Player status.
     mapping(uint256 => address) public indexToPlayer; // Maps a guess index to the player who made the guess.
 
-    /* CONSTRUCTOR */
-
     constructor(uint32 _commitDuration, uint32 _revealDuration) {
```
Comment (Tristan22400) [inline src/Hack.sol:456]:
```suggestion
    constructor(uint32 commitDuration, uint32 revealDuration) {
```

Comment (Tristan22400) [reply src/Hack.sol:456]: On all the exercise. It is the same for the exercise 10.

Comment (makcandrov) [reply src/Hack.sol:456]: Yes that is the tricky case. @MerlinEgalite idk how to name the variable and the parameter in this case (when the variable must be immutable and public): 

```
uint256 public immutable someVariable;

constructor(uint256 _someVariable) {
    someVariable = _someVariable;
}
```

**Review by MerlinEgalite — APPROVED** [review]
Nice!

**Review by Tristan22400 — APPROVED** [review]

**Review by MerlinEgalite — APPROVED** [review]

**Review by Tristan22400 — APPROVED** [review]

**Pull request merged**